src/main/thrift/parquet.thrift

```diff
@@ -237,6 +237,97 @@ struct SizeStatistics {
    3: optional list<i64> definition_level_histogram;
 }
 
+/**
+ * Interpretation for edges of GEOMETRY logical type, i.e. whether the edge
+ * between points represent a straight cartesian line or the shortest line on
+ * the sphere. It applies to all non-point geometry objects.
+ */
+enum Edges {
+  PLANAR = 0;
+  SPHERICAL = 1;
+}
+
+/**
+ * A custom binary-encoded polygon or multi-polygon to represent a covering of
+ * geometries. For example, it may be a bounding box or an envelope of geometries
+ * when a bounding box cannot be built (e.g. a geometry has spherical edges, or if
+ * an edge of geographic coordinates crosses the antimeridian). In addition, it can
+ * also be used to provide vendor-agnostic coverings like S2 or H3 grids.
+ */
+struct Covering {
+  /**
+   * A type of covering. Currently accepted values: "WKB".
+   */
+  1: required string kind;
```

Comment:
Is 'kind' a typical thing in Parquet? If not I'd call this 'type' - kind sounds weird to me, but not sure if 'type' as a very specific meaning for parquet. This is totally just a stylistic thing, so feel free to ignore. But note in the description we say 'a type of covering', not a 'the kind of covering'. Kind seems 'fuzzier', that it's wkb or close to it, while type seems to indicate just one to me.

Reply:
I think there was a discussion on it but I cannot find it. I'm happy to switch `kind` to `type` if it looks better. @jiayuasu @paleolimbot WDYT?

Reply:
I am fine with that but note we also need to change crs kind to be consistent.

Reply:
Is this necessary? It is not in the Iceberg version so I'm wondering if it is something that is required for the initial release.

Reply:
> Is this necessary? It is not in the Iceberg version so I'm wondering if it is something that is required for the initial release.

Yes, it is not used by Iceberg. `Covering` was discussed in https://github.com/apache/parquet-format/pull/240#discussion_r1615980422 to make it flexible enough to add more kinds of geospatial indexes without changing the Parquet spec in the future.

> Parquet already has types (physical and logical). There is a reason for the use of another term here.

Sorry that I cannot find the original discussion. IIRC, `Covering/type` was proposed in the beginning and then switched to `Covering/kind` for similar reasons. So I'll keep using `kind` unless there is a better idea.

Reply:
@rdblue For the WKB geometry encoding, we might introduce vectorized encoding for geometry in the future and allow both WKB and vectorized encoding co-exist. So we want to leave the door open.

Reply:
@jiayuasu I think your last comment for the thread at https://github.com/apache/parquet-format/pull/240#discussion_r1768968271?

Reply:
> So I'll keep using kind unless there is a better idea.

Cool - sorry for the noise, I missed the original discussion to move it to kind.

Reply:
> I am fine with that but note we also need to change crs kind to be consistent.

If this is referencing to the encoding (WKT versus JSON, etc.), I really think that it needs to be renamed "CRS encoding". CRS kind means something else (geographic versus projected versus engineering versus parametric, etc.).